pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -350,10 +350,31 @@ func (c *Cluster) Start() error {
 		return errors.Wrapf(err, "failed to update ceph storage status")
 	}
 
+	// remove the osdBootstrapKey as the osds are running
+	c.deleteOsdBootstrapKeyring()
+
 	logger.Infof("finished running OSDs in namespace %q", namespace)
 	return nil
 }
 
+func (c *Cluster) deleteOsdBootstrapKeyring() {
+	err := cephclient.AuthDelete(c.context, c.clusterInfo, "client.bootstrap-osd")
+	if err != nil {
+		logger.Debugf("failed to delete client.bootstrap-osd bootstrap key in the namespace %q. Error: %v", c.clusterInfo.Namespace, err)
+	} else {
+		logger.Debugf("successfully deleted client.bootstrap-osd bootstrap key in the namespace %q", c.clusterInfo.Namespace)
+	}
+
+	// delete the keyring file
+	bootstrapKeyringPath := "/var/lib/ceph/bootstrap-osd/ceph.keyring"
+	err = os.Remove(bootstrapKeyringPath)
+	if err != nil {
+		logger.Debugf("failed to delete %q file. Error: %v", bootstrapKeyringPath, err)
+	} else {
+		logger.Debugf("successfully deleted %q file", bootstrapKeyringPath)
+	}
```

Comment:
This keyring gets created by/in the prepare pods, but not in the rook operator pod. If the file isn't created in the operator pod, deleting the file won't ever succeed. I think we can just drop this code block